pylib/anki/cards.py

```diff
@@ -89,11 +89,11 @@ def _load_from_backend_card(self, card: cards_pb2.Card) -> None:
         self.odue = card.original_due
         self.odid = anki.decks.DeckId(card.original_deck_id)
         self.flags = card.flags
-        self.data = card.data
+        self.original_position = card.original_position.val
```

Comment:
Am I correct that the Python implementation will default to `original_position=UInt32(val=0)`? That would defeat the purpose of using the generic wrapper.

Reply:
> Presumably it would make sense to have self.card_position being Optional[int]; for that,

Yes, that's what I'm looking for. However, `card.original_position` is always truthy, even if we pass `original_position: None` in Rust. In that case, `card.original_position.val` gives 0, indistinguishable from `original_position: Some(pb::generic::UInt32 { val: 0 })`.
The only solution seems to be the `HasField()` method.

I'll adopt what you've written for the reverse case. Not sure if what I had written is valid as well, or if it only works for oneofs.